RInAppMessaging/Classes/Repositories/CampaignRepository.swift

```diff
@@ -109,11 +112,22 @@ internal class CampaignRepository: CampaignRepositoryType {
         return updateImpressionsLeftInCampaign(campaign, newValue: campaign.impressionsLeft + 1)
     }
 
-    func loadCachedData() {
-        let cachedData = userDataCache.getUserData(identifiers: preferenceRepository.getUserIdentifiers())?.campaignData
-        campaigns.set(value: cachedData ?? [])
+    func loadCachedData(syncWithDefaultUserData: Bool) {
+        var cachedData = userDataCache.getUserData(identifiers: preferenceRepository.getUserIdentifiers())?.campaignData ?? []
+        if syncWithDefaultUserData {
+            userDataCache.getUserData(identifiers: defaultCacheUser)?.campaignData?.forEach({ defaultUserCampaign in
+                if let existingCampaignIndex = cachedData.firstIndex(where: { $0.id == defaultUserCampaign.id }) {
+                    cachedData[existingCampaignIndex] = defaultUserCampaign
```

Comment:
So where there's already an existing campaign in the default user's cached data we replace the newly logged in user's campaign with that existing campaign?

Reply:
Yes. I hope that logic doesn't have any holes